in_toto/matchrule_validators.py

```diff
@@ -37,46 +38,51 @@ def _validate_match_rule(keywords):
         step) = keywords
   else:
     raise securesystemslib.exceptions.FormatError("Wrong rule format,"
-      " should be: MATCH (MATERIAL/PRODUCT)"
+      " should be: MATCH (MATERIAL | PRODUCT)"
       " <path_pattern> [AS <target_path_pattern>] FROM <step>.\n\t"
-      "Got: {}".format(" ".join(keywords)))
+      "Got: {}".format(keywords))
 
   if rule != "MATCH" and rule.upper() != "MATCH":
-    raise securesystemslib.exceptions.FormatError(
-        "Wrong rule to verify! {}".format(rule))
+    raise securesystemslib.exceptions.FormatError("Wrong rule format.\n\t"
+        "Got: {}".format(keywords))
 
   if from_keyword != "FROM" and from_keyword.upper() != "FROM":
-    raise securesystemslib.exceptions.FormatError(
-        "FROM should come before step")
+    raise securesystemslib.exceptions.FormatError("Wrong rule format,"
+        " FROM should come before step.\n\t"
+        "Got: {}".format(keywords))
 
   if as_keyword != "AS" and as_keyword.upper() != "AS":
-    raise securesystemslib.exceptions.FormatError(
-        "AS should come after the step name")
+    raise securesystemslib.exceptions.FormatError("Wrong rule format,"
+        "AS should come after the step name.\n\t"
+        "Got: {}".format(keywords))
 
   if artifact not in MATERIAL_OR_PRODUCT and \
       artifact.upper() not in MATERIAL_OR_PRODUCT:
-    raise securesystemslib.exceptions.FormatError(
-        "Target should be either MATERIAL or PRODUCT!")
+    raise securesystemslib.exceptions.FormatError("Wrong rule format,"
+        " target artifact type should be either MATERIAL or PRODUCT.\n\t"
+        "Got: {}".format(keywords))
 
 def _validate_generic_rule(keywords):
   """ private helper that verifies the syntax of the other rules """
 
   VALID_OPERATIONS = {'CREATE', 'MODIFY', 'DELETE',}
 
   if not isinstance(keywords, list):
-    raise securesystemslib.exceptions.FormatError(
-        "this matching rule is not a list")
+    raise securesystemslib.exceptions.FormatError("Wrong rule format,"
+        " rule must be a list.\n\t"
+        "Got: {}".format(keywords))
 
   if len(keywords) != 2:
-    raise securesystemslib.exceptions.FormatError("Wrong rule format")
+    raise securesystemslib.exceptions.FormatError("Wrong rule format.\n\t"
+        "Got: {}".format(keywords))
```

Comment:
It would help the user more if the exception message emphasized that len(keywords) must be 2, like you do with the others.